db/migrate/20200803113903_migrate_hos_type_in_host_config_groups.foreman_puppet_enc.rb

```diff
@@ -0,0 +1,5 @@
+class MigrateHosTypeInHostConfigGroups < ActiveRecord::Migration[6.0]
+  def change
+    HostConfigGroups.where(:host_type => 'Host::Managed/Base').update_all(:host_type => 'ForemanPuppetEnc::HostFacet')
```

Comment:
I believe this won't be as easy, because we need to change the id column as well. Now we would assume, that the same `host_id` is referencing to the `facet_table`, but that is not the case.
Here is proposed (pseudo = untested) code:
```suggestion
  def up
    group_ids = HostConfigGroups.where(host_type: w%[Host::Base Host::Managed]).pluck(:config_group_id, :host_id)
    group_ids = group_ids.each_with_object(Hash.new([])) {|(cfg_id, host_id), memo| memo[host_id] += [cfg_id] }
    group_ids.each do |host_id, cfg_ids| do
      facet = ForemanPuppetEnc::HostFacet.create(host_id: host_id)
      HostConfigGroups.where(host_type: w%[Host::Base Host::Managed], host_id: host_id).update_all(host_type: 'ForemanPuppetEnc::HostFacet', host_id: facet.id)
    end
  end
```